library/core/src/slice/mod.rs

```diff
@@ -1207,6 +1210,96 @@ impl<T> [T] {
         RChunksExactMut::new(self, chunk_size)
     }
 
+    /// Returns an iterator over the slice producing non-overlapping runs
+    /// of elements using the predicate to separate them.
```

Comment:
I feel like this should have a stronger mark on the fact that it only groups **consecutive** items, especially with such an ambiguous function name.

Reply:
I'd say 'runs of elements' already implies that. But I'd be happy to review your PR if you have an idea on how to improve the wording. Improvements to documentation are always welcome. :)